### ISSUE TITLE
NonBooleanPropertyPrefixedWithIs is catching quasi-boolean properties

### ISSUE THREAD

**Issue by scottkennedy** [detekt/detekt#5362]
Using https://github.com/JetBrains/Exposed, I'm running into a lot of issues with db tables where I have something defined like:
`val isVerified = bool("is_verified")`, where the type is a `Column<Boolean>`.

It's not "technically" a `Boolean`, but it's used in all the same ways.

I'm not sure the best approach here (I'd like to just have an (optional) blanket exclusion for `Column<Boolean>` types, but I'm not sure if that's feasible).  It just doesn't seem to make sense to have a suppression everywhere I refer to a boolean database column, nor do I want to start naming things `columnIsVerified` (any more than I want `intAmount` or `stringName`).

## Expected Behavior
NonBooleanPropertyPrefixedWithIs negative

## Current Behavior
NonBooleanPropertyPrefixedWithIs positive

**Comment by cortinico** [comment]
> a lot of issues with db tables where I have something defined like:

I assume your DB Tables are all defined in one or more file right? This feels like a valid scenario for a local suppression at the file level (or you can use the exclude feature in the config to exclude a `.database` package from being inpsected for this rule).

**Comment by scottkennedy** [comment]
They're declared in many files, throughout many modules (each module generally represents a feature, that contains its own db logic).  `**/database/*.kt` would cover them all though.

But there are also various other functions in those files that I don't want to suppress it on.  If it's not a property of type `Column<Boolean>`, I don't want it suppressed.

**Comment by schalkms** [comment]
I would stay away from putting framework specifics into this rule in general.
Since detekt tries to be framework-agnostic, I'd copy this rule into a custom rule and incorporate the required changes.

**Comment by cortinico** [comment]
> But there are also various other functions in those files that I don't want to suppress it on. If it's not a property of type `Column<Boolean>`, I don't want it suppressed.

I see. This feels like the scenario for a custom rule then as suggested by @schalkms

**Comment by samuelneff** [comment]
What about an option on the rule to allow other classes to be treated as boolean? like

```yml
  NonBooleanPropertyPrefixedWithIs:
    allowedTypes:
      - Column<Boolean>
      - MyBooleanWrapper
```

**Comment by cortinico** [comment]
> What about an option on the rule to allow other classes to be treated as boolean? like

I'm unsure about the value here as it would add unnecessary complexity (you need to handle fully qualified types, generics, and types like `MyColumn<MyBooleanType>`) imho

**Comment by schalkms** [comment]
This feature won't be tackled for the reasons mentioned above.
If someone wants to implement this feature, then it should be included in a custom rule, which we gladly include in detekt's market place.

**Comment by eygraber** [comment]
Would `AtomicBoolean` be an exception, or fall under the same reasons to not special case it?
